qlib/backtest/position.py

```diff
@@ -0,0 +1,456 @@
+        """
+        calculate the value of the all assets except cash in the position
+
+        Returns
+        -------
+        float:
+            the value(money) of all the stock
+        """
+        raise NotImplementedError(f"Please implement the `calculate_stock_value` method")
+
+    def get_stock_list(self) -> List:
+        """
+        Get the list of stocks in the position.
+        """
+        raise NotImplementedError(f"Please implement the `get_stock_list` method")
+
+    def get_stock_price(self, code) -> float:
+        """
+        get the latest price of the stock
+
+        Parameters
+        ----------
+        code :
+            the code of the stock
+        """
+        raise NotImplementedError(f"Please implement the `get_stock_price` method")
+
+    def get_stock_amount(self, code) -> float:
+        """
+        get the amount of the stock
+
+        Parameters
+        ----------
+        code :
+            the code of the stock
+
+        Returns
+        -------
+        float:
+            the amount of the stock
+        """
+        raise NotImplementedError(f"Please implement the `get_stock_amount` method")
+
+    def get_cash(self) -> float:
+        """
+
+        Returns
+        -------
+        float:
+            the cash in position
+        """
+        raise NotImplementedError(f"Please implement the `get_cash` method")
+
+    def get_stock_amount_dict(self) -> Dict:
+        """
+        generate stock amount dict {stock_id : amount of stock}
+
+        Returns
+        -------
+        Dict:
+            {stock_id : amount of stock}
+        """
+        raise NotImplementedError(f"Please implement the `get_stock_amount_dict` method")
+
+    def get_stock_weight_dict(self, only_stock: bool = False) -> Dict:
+        """
+        generate stock weight dict {stock_id : value weight of stock in the position}
+        it is meaningful in the beginning or the end of each trade date
+
+        Parameters
+        ----------
+        only_stock : bool
+            If only_stock=True, the weight of each stock in total stock will be returned
+            If only_stock=False, the weight of each stock in total assets(stock + cash) will be returned
+
+        Returns
+        -------
+        Dict:
+            {stock_id : value weight of stock in the position}
+        """
+        raise NotImplementedError(f"Please implement the `get_stock_weight_dict` method")
+
+    def add_count_all(self, bar):
+        """
+        Will be called at the end of each bar on each level
+
+        Parameters
+        ----------
+        bar :
+            The level to be updated
+        """
+        raise NotImplementedError(f"Please implement the `add_count_all` method")
+
+    def update_weight_all(self):
+        """
+        Updating the position weight;
+
+        # TODO: this function is a little weird. The weight data in the position is in a wrong state after dealing order
```

Comment:
Maybe we can make a more accurate definition for the state of the position.
Make sure users know what is reliable and when it is reliable